### DISCUSSION TITLE
Just wanted to say thanks

### DISCUSSION THREAD

**Discussion by TheyCallMeJames** [philosowaffle/peloton-to-garmin#242] [Show and tell]
Hey @philosowaffle,

I just wanted to say thanks for publishing such an awesome tool.  My girlfriend and I are long time Garmin users so you can imagine we were pretty disappointed when we learned there was no out-of-the-box way to connect the two platforms.  I'll also add that your tool gave me the motivation to finally learn Docker basics (something I have been putting off for some time) and to blow the dust off my Raspberry Pi.

After getting the docker container running on my Pi, I took it a step further and added a push button via the Pi's GPIO pins and a simple bread broad that when pressed, will run the sync.  Now we get that satisfaction of hopping off the bike and "submitting" our hard work by pushing the button for added enjoyment.

![2022-03-26 21_26_38-rapi2_pelotonjpg](https://user-images.githubusercontent.com/10344707/160262890-c8d41258-692e-4e7d-a9b9-6e837f32354a.png)

Thank you so much for everything you've done!!

JP

**Comment by philosowaffle** [comment]
This is such a cool idea!  Thank you for sharing!

I think your next iteration needs to include some flashing lights and a house-wide announcement "_____ has completed a workout" :P

**Comment by jheizer** [comment]
Nice.  I like the button.  I've tied mine into homeassisant and when it detects I've finished a ride it fires off the sync.

**Reply by philosowaffle** [reply]
Fellow HA user, what are you using to detect the ride is finished?

**Reply by blissend** [reply]
Yes how do you detect that?
